neurovault/apps/statmaps/management/commands/benchmark.py

```diff
@@ -0,0 +1,104 @@
+    def __init__(self, timer=None, disable_gc=False, verbose=True):
+        if timer is None:
+            timer = timeit.default_timer
+        self.timer = timer
+        self.disable_gc = disable_gc
+        self.verbose = verbose
+        self.start = self.end = self.interval = None
+    def __enter__(self):
+        if self.disable_gc:
+            self.gc_state = gc.isenabled()
+            gc.disable()
+        self.start = self.timer()
+        return self
+    def __exit__(self, *args):
+        self.end = self.timer()
+        if self.disable_gc and self.gc_state:
+            gc.enable()
+        self.interval = self.end - self.start
+        if self.verbose:
+            print('time taken: %f seconds' % self.interval)
+
+class Command(BaseCommand):
+    args = '<times_to_run>'
+    help = 'bench'
+
+    def handle(self, *args, **options):
+        # TODO: check if /bench folder exists. If not, donwload the full dataset.
+        clearDB()
+        app_path = '/code/neurovault/apps/statmaps/tests'
+        u1 = User.objects.create(username='neurovault3')
+        comparisonCollection1 = Collection(name='comparisonCollection1', owner=u1,
+                                                DOI='10.3389/fninf.2015.00008')
+        comparisonCollection1.save()
+
+        save_statmap_form(
+            image_path=os.path.join(app_path, 'bench/unthres/0003.nii.gz'),
+            collection=comparisonCollection1,
+            image_name="image1",
+            ignore_file_warning=True)
+
+        num_files = len(os.listdir(os.path.join(app_path, 'bench/unthres/')))
+        index_table = np.zeros(num_files)
+        query_table = np.zeros(num_files)
+
+        if args and args[0].isdigit():
+            loop = int(args[0])
+        else:
+            loop = 1
+
+        print "Calculating benchmark every", loop, "images"
+
+        for i, file in enumerate(os.listdir(os.path.join(app_path, 'bench/unthres/'))):
+            print 'Adding subject ' + file
+
+            randomCollection = Collection(name='random' + file, owner=u1, DOI='10.3389/fninf.2015.00008' + str(i))
+            randomCollection.save()
+
+            if i > 0 and i % loop == 0:
+                t = Timer()
+                with t:
+                    image = save_statmap_form(
+                        image_path=os.path.join(app_path, 'bench/unthres/', file),
+                        collection=randomCollection,
+                        image_name=file,
+                        ignore_file_warning=True)
+                index_table[i] = t.interval
+                np.save(os.path.join(app_path, 'bench/results_index_busy'), index_table)
+
+                t = Timer()
+                with t:
+                    _dummy = get_existing_comparisons(image.pk).extra(
+                        select={"abs_score": "abs(similarity_score)"}).order_by("-abs_score")[
+                             0:100]  # "-" indicates descending # TODO: change this depending on the query function
```

Comment:
The way you benchmark querying right now will require you to change this line for the new implemetation. It would be more robust if you would just time how long does it take to generate `images/{pk}/find_similar` page instead.